### PR TITLE
Part of online-store microservice for Comment entity added

### PR DESCRIPTION
**Pull request by groyvstreet**
Part of online-store microservice for Comment entity added

### CODE REVIEW

File: ModsenOnlineStore.Store.API/AutoMapperProfile.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using AutoMapper;
+using ModsenOnlineStore.Store.Domain.DTOs.CommentDTOs;
+using ModsenOnlineStore.Store.Domain.Entities;
+
+namespace ModsenOnlineStore.Store.API
+{
+    public class AutoMapperProfile : Profile
+    {
+        public AutoMapperProfile()
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.API/AutoMapperProfile.cs]: CommentProfile

File: ModsenOnlineStore.Store.API/Controllers/CommentController.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using ModsenOnlineStore.Store.Application.Interfaces.CommentInterfaces;
+using ModsenOnlineStore.Store.Domain.DTOs.CommentDTOs;
+
+namespace ModsenOnlineStore.Store.API.Controllers
+{
+    [Route("[controller]")]
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.API/Controllers/CommentController.cs]: api/[controller]

File: ModsenOnlineStore.Store.API/Controllers/CommentController.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    [ApiController]
+    public class CommentController : ControllerBase
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.API/Controllers/CommentController.cs]: Comment**s**Controller

File: ModsenOnlineStore.Store.Application/Interfaces/CommentInterfaces/ICommentRepository.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace ModsenOnlineStore.Store.Application.Interfaces.CommentInterfaces
+{
+    public interface ICommentRepository
+    {
+        Task<List<Comment>> GetAllComments();
+        Task<Comment?> GetCommentById(int id);
+        Task<List<Comment>> AddComment(Comment product);
+        Task<List<Comment>> UpdateComment(Comment product);
+        Task<List<Comment>> RemoveCommentById(int id);
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.Application/Interfaces/CommentInterfaces/ICommentRepository.cs]: please add empty lines between declarations

File: ModsenOnlineStore.Store.Infrastructure/Services/CommentServices/CommentService.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        {
+            /*var product = await productRepository.GetProductById(updateCommentDto.ProductId);
+
+            if (product is null)
+            {
+                var response = await GetAllComments();
+                response.Success = false;
+                response.Message = "no such product";
+                return response;
+            }*/
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.Infrastructure/Services/CommentServices/CommentService.cs]: delete comments

File: ModsenOnlineStore.Store.API/AutoMapperProfile.cs (hunk 1 above)
Comment (groyvstreet) [reply ModsenOnlineStore.Store.API/AutoMapperProfile.cs]: fixed

File: ModsenOnlineStore.Store.API/Controllers/CommentController.cs (hunk 2 above)
Comment (groyvstreet) [reply ModsenOnlineStore.Store.API/Controllers/CommentController.cs]: fixed

File: ModsenOnlineStore.Store.API/Controllers/CommentController.cs (hunk 3 above)
Comment (groyvstreet) [reply ModsenOnlineStore.Store.API/Controllers/CommentController.cs]: fixed

File: ModsenOnlineStore.Store.Application/Interfaces/CommentInterfaces/ICommentRepository.cs (hunk 4 above)
Comment (groyvstreet) [reply ModsenOnlineStore.Store.Application/Interfaces/CommentInterfaces/ICommentRepository.cs]: fixed

File: ModsenOnlineStore.Store.Infrastructure/Services/CommentServices/CommentService.cs (hunk 5 above)
Comment (groyvstreet) [reply ModsenOnlineStore.Store.Infrastructure/Services/CommentServices/CommentService.cs]: fixed

File: ModsenOnlineStore.Store.Application/Services/CommentServices/CommentService.cs [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            await commentRepository.UpdateComment(comment);
+
+            return new ResponseInfo<string>(data: "updated successfully", success: true, message: "comment");
+        }
+
+        public async Task<ResponseInfo<string>> RemoveCommentById(int id)
+        {
+            await commentRepository.RemoveCommentById(id);
+
+            return new ResponseInfo<string>(data: "removed successfully", success: true, message: "comment");
```
Comment (elv1sabene) [inline ModsenOnlineStore.Store.Application/Services/CommentServices/CommentService.cs]: what is the purpose of such data and message
`success: true` already tells us that object was deleted
`message` it's a bit of an abstract name: is it failure message or some other kind of message?
make another ResponseInfo class without any data and non-generic so it store success flag and message

Comment (groyvstreet) [reply ModsenOnlineStore.Store.Application/Services/CommentServices/CommentService.cs]: NoDataResponseInfo class was created, now success flag can be true or false according to the result of operation, message tells that operation done or purpose of operation failure

Comment (elv1sabene) [reply ModsenOnlineStore.Store.Application/Services/CommentServices/CommentService.cs]: It is already clear that operation is done when `success = true`. So I think that the only one reason to have this message is failure message

**Review by elv1sabene — APPROVED** [review]

**Pull request merged**